feature/preview/src/main/java/com/google/jetpackcamera/feature/preview/quicksettings/ui/QuickSettingsComponents.kt

```diff
@@ -441,6 +453,88 @@ fun ToggleQuickSettingsButton(
 
 // subcomponents used to build completed components
 
+@Composable
+fun QuickSettingCarouselButton(
+    modifier: Modifier = Modifier,
+    enum: QuickSettingsEnum,
+    onClick: () -> Unit,
+    isHighLighted: Boolean = false,
+    enabled: Boolean = true
+) {
+    QuickSettingCarouselButton(
+        modifier = modifier,
+        text = stringResource(id = enum.getTextResId()),
+        accessibilityText = stringResource(id = enum.getDescriptionResId()),
+        onClick = { onClick() },
+        isHighlighted = isHighLighted,
+        enabled = enabled,
+        painter = enum.getPainter()
+    )
+}
+
+/**
+ *
+ * @param isHighlighted true if the button is currently checked; false otherwise.
+ * @param onClick will be called when the user clicks the button.
+ * @param text The text label to display below the icon.
+ * @param painter The icon to display inside the button.
+ * @param modifier The Modifier to be applied to the button.
+ */
+@OptIn(ExperimentalMaterial3ExpressiveApi::class)
+@Composable
+fun QuickSettingCarouselButton(
+    onClick: () -> Unit,
+    text: String,
+    accessibilityText: String,
+    painter: Painter,
+    modifier: Modifier = Modifier,
+    isHighlighted: Boolean = false,
+    enabled: Boolean = true // todo: handle hidden disabled or visible disabled
+) {
+    // todo(kc): better sizing
+    val iconSize = dimensionResource(id = R.dimen.quick_settings_ui_item_icon_size)
+
+    Column(
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.Center
+    ) {
+        ToggleButton(
+            checked = isHighlighted,
+            enabled = enabled,
+            onCheckedChange = { _ -> onClick() },
+            // 1. Size updated to width 48.dp and height 56.dp
+            modifier = modifier.size(width = 96.dp, height = 112.dp),
+            shapes = ToggleButtonDefaults.shapes()
+                .copy(
+                    checkedShape = RoundedCornerShape(percent = 50),
+                    shape = RoundedCornerShape(percent = 30)
+                ),
+
+            // fixme(kc): the colors of disabled buttons here appear invisible when light mode is enabled
+            colors = ToggleButtonDefaults.toggleButtonColors(
+                containerColor = Color.White.copy(alpha = 0.16F),
+                contentColor = MaterialTheme.colorScheme.secondaryFixed,
+                checkedContainerColor = MaterialTheme.colorScheme.primaryFixedDim,
+                checkedContentColor = MaterialTheme.colorScheme.onPrimaryFixed
+            )
+        ) {
+            Icon(
+                modifier = Modifier.size(iconSize),
+                painter = painter,
+                contentDescription = accessibilityText
+            )
+        }
+
+        Spacer(Modifier.height(4.dp))
+        Text(
+            text = text,
+            color = Color.White,
+            maxLines = 1,
+            overflow = TextOverflow.Ellipsis
```

Comment:
From the mocks, it looks like we should be soft wrapping rather than only allowing 1 line and overflowing. Or maybe the maxLines should be 2?